src/api/File.js

```diff
@@ -32,34 +34,56 @@ class File extends Item {
      * API URL for files
      *
      * @param {string} [id] - Optional file id
+     * @param {string} [versionId] - Optional file version id
      * @return {string} base url for files
      */
-    getUrl(id: string): string {
-        const suffix: string = id ? `/${id}` : '';
-        return `${this.getBaseApiUrl()}/files${suffix}`;
+    getUrl(id: string, versionId: ?string): string {
+        const fileSuffix: string = id ? `/${id}` : '';
+        const fileVersionSuffix: string = versionId ? `/versions/${versionId}` : '';
+        return `${this.getBaseApiUrl()}/files${fileSuffix}${fileVersionSuffix}`;
     }
 
     /**
      * API for getting download URL for files
      *
      * @param {string} id - File id
+     * @param {string|null} versionId - File version id
      * @param {Function} successCallback - Success callback
      * @param {Function} errorCallback - Error callback
      * @return {void}
      */
-    getDownloadUrl(id: string, successCallback: Function, errorCallback: ElementsErrorCallback): Promise<void> {
+    getDownloadUrl(
+        id: string,
+        versionId: ?string,
+        successCallback: Function,
+        errorCallback: ElementsErrorCallback,
+    ): Promise<void> {
         this.errorCode = ERROR_CODE_GET_DOWNLOAD_URL;
-        this.successCallback = successCallback;
         this.errorCallback = errorCallback;
+        this.successCallback = successCallback;
+
         return this.xhr
             .get({
-                url: this.getUrl(id),
+                url: this.getUrl(id, versionId),
```

Comment:
The relationship between the `File` and `Versions` API modules is a little awkward. I think the ideal solution would be to introduce a `FileVersion` module that extends `File` and shares the base download logic. Not sure I want to tackle that right now, though.